### PR TITLE
User features

### PR DESCRIPTION
**Pull request by BenSheridanEdwards**

### CODE REVIEW

**Comment by jj49411** [comment]
please merge

**Comment by jj49411** [comment]
merge please

**Review by jj49411 — COMMENTED** [review]
looks fine

**Review by jj49411 — COMMENTED** [review]
please merge

File: Gemfile [hunk 1]
```diff
@@ -5,6 +5,7 @@ git_source(:github) do |repo_name|
   "https://github.com/#{repo_name}.git"
 end
 
+gem "factory_bot_rails"
```
Comment (Hyan18) [inline Gemfile:8]: gem for passing user to controller tests

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -12,4 +12,12 @@ def require_login
     redirect_to '/login' unless current_user
   end
 
+  # def log_in(user)
```
Comment (Hyan18) [inline app/controllers/application_controller.rb:15]: not used yet, for future

File: app/helpers/user_helper.rb [hunk 3]
```diff
@@ -1,2 +1,18 @@
-module UserHelper
-end
+ module UserHelper
+#   def signup
```
Comment (Hyan18) [inline app/helpers/user_helper.rb]: moved to web_helper, should delete

File: spec/controllers/sessions_controller_spec.rb [hunk 4]
```diff
@@ -9,4 +9,18 @@
     end
   end
 
+  # describe "GET #create" do
```
Comment (Hyan18) [inline spec/controllers/sessions_controller_spec.rb:12]: to be edited

File: spec/helpers/user_helper_spec.rb [hunk 5]
```diff
@@ -1,15 +1,10 @@
 require 'rails_helper'
 
-# Specs in this file have access to a helper object that includes
-# the UserHelper. For example:
-#
-# describe UserHelper do
-#   describe "string concat" do
-#     it "concats two strings with spaces" do
-#       expect(helper.concat_strings("this","that")).to eq("this that")
+# RSpec.describe UserHelper, type: :helper do
```
Comment (Hyan18) [inline spec/helpers/user_helper_spec.rb:3]: Can be deleted as moved to web_helper

File: spec/spec_helper.rb [hunk 6]
```diff
@@ -20,15 +23,14 @@
 
 RSpec.configure do |config|
 
-  RSpec::Matchers.define :appear_before do |later_content|
-    match do |earlier_content|
-      page.body.index(earlier_content) < page.body.index(later_content)
-    end
-  end  
+  config.include UserHelper, :type => :feature
```
Comment (Hyan18) [inline spec/spec_helper.rb:32]: Can get rid of this as user helper moved to web_helpers

**Review by Hyan18 — APPROVED** [review]

**Pull request merged**